### ISSUE TITLE
webpack.module.loaders breaks the build

### ISSUE THREAD

**Issue by eabay** [grommet/grommet-toolbox#1]
Liked the toolbox but got a problem :)

Here is my gulpfile.js:

```js
var gulp = require('gulp');
var path = require('path');
var fs = require('fs');
var grommetToolbox = require('grommet-toolbox').default;

var alias = p => path.resolve(__dirname, 'node_modules', p);

var opts = {
  base: '.',
  dist: path.resolve(__dirname, 'dist/'),
  copyAssets: [
    'src/index.html',
    { asset: 'src/img/**', dist: 'dist/img/' }
  ],
  scssAssets: ['src/scss/**/*.scss'],
  jsAssets: ['src/js/**/*.js'],
  mainJs: 'src/js/index.js',
  mainScss: 'src/scss/index.scss',
  webpack: {
    resolve: {
      root: ['src/js', 'src/scss', 'node_modules'].map(p => path.resolve(__dirname, p)),
      alias: {
        'grommet-index/scss': alias('grommet-index/src/scss'),
        'grommet-index': alias('grommet-index/src/js'),
        'grommet/scss': alias('grommet/src/scss'),
        'grommet': alias('grommet/src/js')
      }
    },
    module: {
      loaders: [{
        test: /\.svg$/,
        loader: 'file-loader?mimetype=image/svg+xml'
      }]
    }
  },
  devServerPort: 4000,
  devServerProxy: {
    '/api/*': {
      target: process.env.SERVER_URL,
      rewrite: function(req) {
        req.url = req.url.replace(/^\/api/, '');
      }
    }
  }
};

grommetToolbox(gulp, opts);
```

There is a file loader for `.svg` files but webpack does not produce a healthy built.

Console output:

```shell
[01:08:20] Using gulpfile ~/workspace/projects/bigs/services/uinext/gulpfile.js
[01:08:20] Starting 'dev-preprocess'...
[01:08:20] Starting 'preprocess'...
[01:08:20] Starting 'clean'...
[01:08:20] Finished 'clean' after 6.38 ms
[01:08:20] Starting 'copy'...
[01:08:20] Finished 'copy' after 17 ms
[01:08:20] Starting 'generate-icons'...
[01:08:20] Finished 'generate-icons' after 32 μs
[01:08:20] Starting 'jslint'...
[01:08:23] Finished 'jslint' after 2.65 s
[01:08:23] Starting 'scsslint'...
[scsslint] scsslint skipped!
[scsslint] scss-lint is not installed. Please install ruby and the ruby gem scss-lint.
[01:08:23] Finished 'scsslint' after 1.15 ms
[01:08:23] Finished 'preprocess' after 2.67 s
[01:08:23] Finished 'dev-preprocess' after 2.67 s
[01:08:23] Starting 'dev'...
[01:08:23] Finished 'dev' after 137 ms
[webpack-dev-server] started: opening the app in your default browser...
[01:08:23] Opening http://localhost:4000/webpack-dev-server/ using the default OS app
Hash: 894c5fada0850735885b
Version: webpack 1.12.14
Time: 1131ms
                              Asset     Size  Chunks             Chunk Names
d38545ef77a2ad664a19320f7d246218.js  1.95 kB          [emitted]  
                           index.js   274 kB       0  [emitted]  app
chunk    {0} index.js (app) 216 kB [rendered]
    [0] multi app 52 bytes {0} [built]
    [1] (webpack)-dev-server/client?http://localhost:4000 2.67 kB {0} [built]
    [2] ./~/url/url.js 22.3 kB {0} [built]
    [3] ./~/punycode/punycode.js 14.7 kB {0} [built]
    [4] (webpack)/buildin/module.js 251 bytes {0} [built]
    [5] ./~/querystring/index.js 127 bytes {0} [built]
    [6] ./~/querystring/decode.js 2.4 kB {0} [built]
    [7] ./~/querystring/encode.js 2.09 kB {0} [built]
    [8] ./~/sockjs-client/lib/entry.js 244 bytes {0} [built]
    [9] ./~/sockjs-client/lib/transport-list.js 613 bytes {0} [built]
   [10] ./~/sockjs-client/lib/transport/websocket.js 2.68 kB {0} [built]
   [11] ./~/process/browser.js 2.06 kB {0} [built]
   [12] ./~/sockjs-client/lib/utils/event.js 2 kB {0} [built]
   [13] ./~/sockjs-client/lib/utils/random.js 746 bytes {0} [built]
   [14] ./~/sockjs-client/lib/utils/browser-crypto.js 438 bytes {0} [built]
   [15] ./~/sockjs-client/lib/utils/url.js 975 bytes {0} [built]
   [16] ./~/url-parse/index.js 8.09 kB {0} [built]
   [17] ./~/requires-port/index.js 753 bytes {0} [built]
   [18] ./~/url-parse/lolcation.js 1.58 kB {0} [built]
   [19] ./~/querystringify/index.js 1.3 kB {0} [built]
   [20] ./~/debug/browser.js 3.76 kB {0} [built]
   [21] ./~/debug/debug.js 4.1 kB {0} [built]
   [22] ./~/ms/index.js 2.33 kB {0} [built]
   [23] ./~/inherits/inherits_browser.js 672 bytes {0} [built]
   [24] ./~/sockjs-client/lib/event/emitter.js 1.11 kB {0} [built]
   [25] ./~/sockjs-client/lib/event/eventtarget.js 1.75 kB {0} [built]
   [26] ./~/sockjs-client/lib/transport/browser/websocket.js 58 bytes {0} [built]
   [27] ./~/sockjs-client/lib/transport/xhr-streaming.js 1.25 kB {0} [built]
   [28] ./~/sockjs-client/lib/transport/lib/ajax-based.js 1.31 kB {0} [built]
   [29] ./~/sockjs-client/lib/transport/lib/sender-receiver.js 1.15 kB {0} [built]
   [30] ./~/sockjs-client/lib/transport/lib/buffered-sender.js 2.3 kB {0} [built]
   [31] ./~/sockjs-client/lib/transport/lib/polling.js 1.32 kB {0} [built]
   [32] ./~/sockjs-client/lib/transport/receiver/xhr.js 1.58 kB {0} [built]
   [33] ./~/sockjs-client/lib/transport/sender/xhr-cors.js 343 bytes {0} [built]
   [34] ./~/sockjs-client/lib/transport/browser/abstract-xhr.js 4.67 kB {0} [built]
   [35] ./~/sockjs-client/lib/transport/sender/xhr-local.js 352 bytes {0} [built]
   [36] ./~/sockjs-client/lib/utils/browser.js 560 bytes {0} [built]
   [37] ./~/sockjs-client/lib/transport/xdr-streaming.js 984 bytes {0} [built]
   [38] ./~/sockjs-client/lib/transport/sender/xdr.js 2.42 kB {0} [built]
   [39] ./~/sockjs-client/lib/transport/eventsource.js 766 bytes {0} [built]
   [40] ./~/sockjs-client/lib/transport/receiver/eventsource.js 1.58 kB {0} [built]
   [41] ./~/sockjs-client/lib/transport/browser/eventsource.js 37 bytes {0} [built]
   [42] ./~/sockjs-client/lib/transport/lib/iframe-wrap.js 981 bytes {0} [built]
   [43] ./~/sockjs-client/lib/transport/iframe.js 3.8 kB {0} [built]
   [44] ./~/json3/lib/json3.js 43.3 kB {0} [built]
   [45] (webpack)/buildin/amd-options.js 43 bytes {0} [built]
   [46] ./~/sockjs-client/lib/version.js 25 bytes {0} [built]
   [47] ./~/sockjs-client/lib/utils/iframe.js 4.95 kB {0} [built]
   [48] ./~/sockjs-client/lib/utils/object.js 532 bytes {0} [built]
   [49] ./~/sockjs-client/lib/transport/htmlfile.js 710 bytes {0} [built]
   [50] ./~/sockjs-client/lib/transport/receiver/htmlfile.js 2.17 kB {0} [built]
   [51] ./~/sockjs-client/lib/transport/xhr-polling.js 894 bytes {0} [built]
   [52] ./~/sockjs-client/lib/transport/xdr-polling.js 712 bytes {0} [built]
   [53] ./~/sockjs-client/lib/transport/jsonp-polling.js 1.02 kB {0} [built]
   [54] ./~/sockjs-client/lib/transport/receiver/jsonp.js 5.49 kB {0} [built]
   [55] ./~/sockjs-client/lib/transport/sender/jsonp.js 2.46 kB {0} [built]
   [56] ./~/sockjs-client/lib/main.js 11.9 kB {0} [built]
   [57] ./~/sockjs-client/lib/shims.js 18.2 kB {0} [built]
   [58] ./~/sockjs-client/lib/utils/escape.js 2.31 kB {0} [built]
   [59] ./~/sockjs-client/lib/utils/transport.js 1.35 kB {0} [built]
   [60] ./~/sockjs-client/lib/utils/log.js 386 bytes {0} [built]
   [61] ./~/sockjs-client/lib/event/event.js 485 bytes {0} [built]
   [62] ./~/sockjs-client/lib/location.js 177 bytes {0} [built]
   [63] ./~/sockjs-client/lib/event/close.js 295 bytes {0} [built]
   [64] ./~/sockjs-client/lib/event/trans-message.js 292 bytes {0} [built]
   [65] ./~/sockjs-client/lib/info-receiver.js 2.22 kB {0} [built]
   [66] ./~/sockjs-client/lib/transport/sender/xhr-fake.js 456 bytes {0} [built]
   [67] ./~/sockjs-client/lib/info-iframe.js 1.52 kB {0} [built]
   [68] ./~/sockjs-client/lib/info-iframe-receiver.js 791 bytes {0} [built]
   [69] ./~/sockjs-client/lib/info-ajax.js 1.03 kB {0} [built]
   [70] ./~/sockjs-client/lib/iframe-bootstrap.js 2.9 kB {0} [built]
   [71] ./~/sockjs-client/lib/facade.js 723 bytes {0} [built]
   [72] ./~/strip-ansi/index.js 161 bytes {0} [built]
   [73] ./~/ansi-regex/index.js 135 bytes {0} [built]
   [74] (webpack)/hot/only-dev-server.js 2.25 kB {0} [built]
   [75] (webpack)/hot/log-apply-result.js 813 bytes {0} [built]
   [76] ./src/js/index.js 81 bytes {0} [built]
webpack: bundle is now VALID.
```

When I remove the `module` section from options, it builds but gives errors for svg files.

Same config was working with old utils and as far as I can see, there is no change in how module loaders are handled.

I couldn't figure out where the problem is.

**Comment by alansouzati** [comment]
I'm sorry but I must be missing something.

When you say:

> There is a file loader for .svg files but webpack does not produce a healthy built.

What do you mean by that? 

I could not see anything wrong with your console...

**Comment by eabay** [comment]
Sorry for the poor explanation :(

I installed `grommet-toolbox` and I converted my `gulpfile.js` file content as I shared above but I got a blank screen when I execute `gulp dev`. There was no error on the console output, everything seemed working normal like you said. Then I realized that there is no Grommet component file path or any other file path from application source in the output. I tried `gulp dist` and saw that `mainJs`(*src/js/index.js*) file was there as is, it wasn't processed at all. There must be something with the module loader definitions. When I remove `webpack.module.loaders` from my options, it works but then I have a problem with SVG files since they are not processed without a file loader.

I see that module loader definitions are [merged](https://github.com/grommet/grommet-toolbox/blob/04faa6b3194bfe3fbeef20bc529d4d5fd0ed83b3/src/webpack.dev.config.js#L42-L46), not replaced but seems like they are replaced. I couldn't follow what's going on.
